### ISSUE TITLE
gcc@5 failed to build on 10.13

### ISSUE THREAD

**Issue by gbrand-salesforce** [Homebrew/homebrew-core#26983]
**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [x] are reporting a bug others will be able to reproduce and not asking a question. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh
- [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, official formula (not cask)? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new. If it's a `brew cask` problem please file this issue at https://github.com/caskroom/homebrew-cask/issues/new. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [x] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

To help us debug your issue please explain:
- What you were trying to do (and why)

> Tried installing gcc 5, in order to be able to build xgboost with python binding.

- What happened (include command output)

> ==> Patching
> ==> Applying 5.4.0.patch
> patching file gcc/jit/Make-lang.in
> ==> Applying xcode9.patch
> patching file gcc/system.h
> Hunk #1 succeeded at 228 with fuzz 2 (offset 12 lines).
> ==> Applying 10.13_headers.patch
> patching file fixincludes/fixincl.x
> patching file fixincludes/inclhack.def
> ==> Downloading https://gcc.gnu.org/pub/gcc/infrastructure/isl-0.14.tar.bz2
> Already downloaded: /Users/gbrand/Library/Caches/Homebrew/gcc@5--isl-0.14.tar.bz2
> ==> ../configure --build=x86_64-apple-darwin17.5.0 --prefix=/usr/local/Cellar/gcc@5/5.5.0_2 --libdir=/usr/local/Cellar/gcc@5/5.5.0_2/lib/gcc/5 --enable-languages=c,c++,objc,obj-c++,fortran --program-suffix=-5 --with-gmp=/usr/local/opt/gmp --with-mpfr=/usr/local/opt/mpfr --with-mpc=/usr/local/opt/libmpc --with-system-zlib --enable-libstdcxx
> ==> make bootstrap
> Last 15 lines from /Users/gbrand/Library/Logs/Homebrew/gcc@5/02.make:
> checking whether strverscmp is declared... no
> checking whether canonicalize_file_name must be declared... yes
> checking for stdlib.h... (cached) yes
> checking for unistd.h... (cached) yes
> checking for getpagesize... (cached) yes
> checking for working mmap... yes
> checking for working strncmp... yes
> configure: updating cache ./config.cache
> configure: creating ./config.status
> config.status: creating Makefile
> config.status: creating testsuite/Makefile
> config.status: creating config.h
> config.status: executing default commands
> make[1]: *** [stage2-bubble] Error 2
> make: *** [bootstrap] Error 2
> 

- What you expected to happen

> I would have liked gcc 5 to be installed...

- Step-by-step reproduction instructions (by running `brew install` commands)

> brew install gcc@5

Here's the output of brew-logs gcc@5:
https://gist.github.com/873579e846cb972fcf2fccdd5c90fabd

# Formula additions or changes
To get formulae added or changed in Homebrew please file a [Pull Request](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)
We will close issues requesting formulae changes.

**Comment by fxcoudert** [comment]
Bootstrap is failing during stage 2 configuration of the vendored isl, which cannot find `gmp.h`:

```
checking gmp.h presence... no
checking for gmp.h... no
configure: error: gmp.h header not found
make[2]: *** [configure-stage2-isl] Error 1
```

I've started a Jenkins build to see if we can reproduce the issue: https://jenkins.brew.sh/job/Homebrew%20Testing/752/

**Comment by fxcoudert** [comment]
@gbrand-salesforce should also state that install from our prebuilt bottles will fix this for you :)

**Comment by gbrand-salesforce** [comment]
@fxcoudert Thanks, xcode-select --install did make it get the prebuilt bottle and solved my issue, although it's weird that it didn't find the gmp.h, it is installed in my system, and is in /usr/local/include.

I was sure I already had the command line tools installed already, but apparently, I didn't... I wonder if brew could have issued a warning in brew doctor for that.

**Comment by fxcoudert** [comment]
We can't reproduce this weird issue locally or on our test machines. I'm closing for now, we'll wait and see if other people are affected. Thanks for reporting @gbrand-salesforce!

**Comment by gerd2002** [comment]
I can reproduce this, but I have since retried building it (and cancelled), so I don't have any logs. Should I reproduce the error another time (2h of waiting until it fails)?

**Comment by kaidokert** [comment]
I'm getting this :
```
Kaidos-MacBook-Pro:cpp kaido$ brew update
Already up-to-date.
Kaidos-MacBook-Pro:cpp kaido$ brew upgrade gcc@5
==> Upgrading 1 outdated package, with result:
gcc@5 5.5.0 -> 5.5.0_2
Error: No available formula with the name "isl@0.14" (dependency of gcc@5)
Kaidos-MacBook-Pro:cpp kaido$ brew install isl
Warning: isl 0.19 is already installed and up-to-date
To reinstall 0.19, run `brew reinstall isl`
Kaidos-MacBook-Pro:cpp kaido$ brew list --versions isl
isl 0.18 0.19
```

**Comment by gerd2002** [comment]
@kaidokert yours seems to be a completely different error

**Comment by btracey** [comment]
I'm getting the same error as @kaidokert 

```
brendan:~$ brew update
Already up-to-date.
brendan:~$ brew upgrade gcc@5
==> Upgrading 1 outdated package, with result:
gcc@5 5.4.0_1 -> 5.5.0_2
Error: No available formula with the name "isl@0.14" (dependency of gcc@5)
brendan:~$ brew install isl
Warning: isl 0.19 is already installed and up-to-date
To reinstall 0.19, run `brew reinstall isl`
brendan:~$ brew list --versions isl
isl 0.17.1 0.19 0.18
```

**Comment by fxcoudert** [comment]
After a `brew up`, `brew info gcc@5` should show that it does not have an `isl@0.14` dependency anymore.
